pkg/kotsadmlicense/serviceaccount.go

```diff
@@ -0,0 +1,54 @@
+package license
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+)
+
+type ServiceAccountToken struct {
+	Identity string `json:"i"`
+	Secret   string `json:"s"`
+}
+
+// ValidateServiceAccountToken checks if the service account token is valid and if it matches the current license identity.
+// It returns the service account token, a boolean indicating if the secret has been updated, and an error if any.
+func ValidateServiceAccountToken(token, currentLicenseID string) (*ServiceAccountToken, bool, error) {
+	newToken, err := extractIdentityFromLicenseID(token)
+	if err != nil {
+		return nil, false, fmt.Errorf("failed to extract new token identity: %w", err)
+	}
+
+	currentToken, err := extractIdentityFromLicenseID(currentLicenseID)
```

Comment:
not all license ids are in token format, no?

Reply:
Correct - if your license ID is a classic license ID and not a token, then the button to update the token will not be shown

Reply:
should probably check if it's a service account token similar to how the ui does it?

Reply:
That's what this is doing though? Apparently I need more comments on the functions

Reply:
added a comment on the 'extractIdentityFromLicenseID' function, and then renamed it to 'extractIdentityFromToken'